### ISSUE TITLE
Full support of processing info

### ISSUE THREAD

**Issue by JGuetschow** [primap-community/primap2#280]
**Is your feature request related to a problem? Please describe.**

The csg generated processing information and adds it to the result dataset. Other function currently can't handle this information and will throw an error when called with a dataset including processing information. 

**Describe the solution you'd like**

All functions can handle processing info. Most function only add info about their processing if the input dataset already has processing info. The csg always adds processing info. To add processing info without the csg add a function that initializes empty processing info in a dataset.
  
**Describe alternatives you've considered**

Always add processing info. This saves information that we often don't need.

**Additional context**

TODO: Add a list of functions to modify here:

Some functions might be complex,e.g. merge, where processing information from two datasets has to be integrated into the processing info of the result. Possible solution: keep only info from calling dataset and discard te infor from the dataset thatis merged in.

**Comment by fuchsi-huber** [comment]
I disagree with the "possible solution" mentioned in the last sentence. A merge without processing information about all merged arrays isn't really better than not having it at all: it suggests that the processing that happened could be traced back while that is actually not the case. Same for `sum` and other aggregation.

The cleanest solution imho would be to have `TimeseriesProcessingDescription` not be a `list` but some form of tree structure.
Equivalently adding and `id` and a `parents` field to the processing steps could preserve the `list` data structure but show what actually happened. The merge or sum would then concatenate all processing info lists, ensure they have unique ids (eg. prefix them in some way) and could refer to them as its `parents`. The question is mostly about which is more readable / helpful...

It would look something like this:
```
[
  {"function": "load_CDIAC2023I", "id": "A/0", "parents":[]},
  {"function": "do_something", "id": "A/1", "parents":["A/0"]},
  {"function": "load_CRF2023I", "id": "B/0", "parents":[]},
  {"function": "do_something", "id": "B/1", "parents":["B/0"]},
  {"function": "merge", "description": "merged parent[1] into parent[0]" "id": "0", "parents":["A/1", "B/1"]}
]
```

Opinions on this? Also @mikapfl ?

**Comment by mikapfl** [comment]
True, a tree structure would be more suitable to describe what is happening. A list was fine for the CSG, but even then: what should the CSG do if the input dataset has processing info already? It is a special case of merge, which would be handled elegantly using a tree structure.

How to implement it (id + parents, or something like networkx or so), let's find out what is most useful to work with. So far, we don't have consumers of processing info at all, so maybe we want to hash out the most important consumers of processing info, to see what use cases are easy to solve with a specific implementation.

Consumers I can think of off the top of my head:
* Timeseries-level human-readable processing info.
* Timeseries- and dataset-level generated reference/citation lists.
* Timeseries-level generation of a table with sources and how they were used.
* Timeseries- and dataset-level generated flow diagram of processing.
* Sources for an individual data point, with e.g. evaluation if it is "country reported", "country reported, adjusted", "third party", or "extrapolated" or so. Essentially, putting it into big, interesting bins.

Other things that you'd really like to see, @JGuetschow ?
